### ISSUE TITLE
Crash when mirroring blueprints [10.10]

### ISSUE THREAD

**Issue by Glockshna** [ChiselsAndBits/Chisels-and-Bits#63]
Crash report: http://pastebin.com/7WWTdKq5

Reproduction Steps: https://dl.dropboxusercontent.com/s/cubr8ksr6t2etjt/2016-06-26_01-23-20.webm?dl=0 (Excuse the discord idiots lol)

**Comment by AlgorithmX2** [comment]
10.11 should resolve the issue, I accidentally built it vs 1.10 instead of 1.9.4

**Comment by AlgorithmX2** [comment]
@Glockshna 

I noticed your creation on a youtube video where you were seeing delayed re-rendering of blocks.

I'd like to look into that issue, but it requires a fairly complicated creation. I was curious it it was possible to get a hold of the world, or section you used to in the video which contained the issue so I could properly recreate that problem.

Also if you wouldn't mind what memory settings are you using, and how much ram is generally free in the instance? ( F3 Stats Screen )

At a glance you don't seem to have optifine, but I would also like to confirm that your not using it.

**Comment by Glockshna** [comment]
@AlgorithmX2 I do have optifine and I'll make a short video for you showing the debug screen and a few issues I have related to rendering. I will need to get the world from my buddy who hosts the server. We have a custom launcher for our modpack so if you needed mod context feel free to grab it at http://aiminecraft.com/Configs/AI%20Launcher%20(v0.3).jar It's the Unstable pack option.

My RAM settings are: Minimum: 1024Mb Maximum: 4096Mb

I'll also whitelist you on the server if you want to join it. It should already be in the server list when you open the pack.

Video: https://youtu.be/-ahMszKbz20
Map: http://aiminecraft.com/World_Downloads/map.6.27.zip (600MB+)

**Comment by AlgorithmX2** [comment]
Thanks for the info, I'll see what I can figure out.

**Comment by AlgorithmX2** [comment]
Would it be possible to try a new test build on the server / clients?

https://dl.dropboxusercontent.com/u/9117298/cb/anti-builds/chiselsandbits-10.12-pre1.jar
- requires forge 1.9.4-12.17.0.1976

I was able to reproduce some of the issues, and I think I have made some helpful improvements. 

The issue with the ghost block is something I was unable to reproduce, that seems to be caused by a de-sync between the server and client, and isn't related to the rendering.

However the new build should probably resolve the larger issues with the rendering bugs.

You'll still see the area re-render after you make a change the first time but that can't be avoided, that happens when C&B is switching renderers.

However the large lag when placing blocks, and the issues with blocks hanging out unrendered should mostly be resolved based on my testing at least.

**Comment by Glockshna** [comment]
Tried the new build and ran into another issue which I think is related to C&B but not sure.

I'm still getting occasional ghost blocks even in single player but I can't reproduce them reliably. When I was making a video trying to show this I ran into a CPU usage issue that I hit once before where the client will hitch BAD and CPU usage spikes. The last time this happened I wasn't recording so that's not the cause. I added my task manager to OBS so you can see what's happening.

The hitching when placing blocks seems to have been reduced substantially but it's still there for sure.

https://youtu.be/FNZ_Dgxfcjw?t=200

**Comment by AlgorithmX2** [comment]
The issue your demonstrating might be related to your memory usage reporting 95% I'm not sure what all is using that much memory, but I wasn't experiencing anything like that testing last evening, and I had significantly less memory then your setup ( 2g ), however I was only using C&B and FCB in my instance so there is an expected difference. However if your game is nearing max memory it will cause substantial pauses while trying to free up memory to keep the program alive.

I'm not sure why your F3 isn't showing the two memory values like it normally does. ( maybe optifine? maybe try without it, not sure. )

Its much harder to get a read on your memory situation without the current and max values that vanilla normally displays.

The changes I made shouldn't have caused that sort of effect ( and didn't for me )

I removed a few causes where C&B was re-rendering to many blocks, and added some settings to control geometry uploads, you can try reducing Client Performance->Max Milliseconds Per Frame to 1ms instead of the default value it used to be 1ms, I raised it to try and reduce the delay between the re-render and the blocks being visible, but that should not be causing consistent pausing like you displayed in the video, if anything it would only happen on initial conversion to the dynamic renderer.

I would suggest trying to adjust that setting, maybe try without Optifine, possible get a video of how the normal vanilla memory display looks without it being changed.

In short based on what I've seen so far It seems to be a memory related issue, however whats causing it remains to be seen.

**Comment by Glockshna** [comment]
I wasn't suggesting your changes caused this. The first time it happened was several days ago before my first report even. Just figured I'd show it to you just in case you had any insight since I happened to be recording when it happened. A restart of the game normally fixes that hitching issue as shown in the video.

**Comment by AlgorithmX2** [comment]
Ok, I thought you were saying that that issue was new.

That further implys that its a memory related issue, something is likely slowly using more memory while your playing, which eventually reaches the cap, and then causes that issue.

You can use http://www.eclipse.org/mat/ to take a heap dump and analyze the data looking for "Leak Suspects" this isn't full proof, but it might give you some idea of what the cause of the issue is. If your uncertain of the results you can send me some screen shots of the result of that report and we can discuss them to see if it is a real issue or not.

Sometimes the strangest things can cause issues, only solution is to analyze the problem and see what we find.

**Comment by Glockshna** [comment]
For sure. Gonna be taking a break from minecraft for a couple days to get some Factorio in. I'll have a look at that when I get back to MC since it's not super urgent.
